plugin-copilot/src/main/java/appland/copilotChat/copilot/GitHubCopilotService.java

```diff
@@ -52,13 +54,75 @@ public boolean isCopilotAuthenticated() {
         return registry.getEncoding(encodingType);
     }
 
+    /**
+     * Whether the content exclusions have been downloaded.
+     */
+    private boolean contentExclusionsDownloaded = false;
+
+    /**
+     * Downloads the content exclusions from the GitHub Copilot API and writes them to the global ignores file.
+     */
+    private void downloadContentExclusions(String githubToken) throws IOException {
+        var exclusions = CopilotContentExclusion.fetchGlobal(githubToken);
+        var paths = new ArrayList<String>();
+        for (var exclusion : exclusions) {
+            LOG.debug("Downloaded content exclusion: " + exclusion);
+            for (var rule : exclusion.rules())
+                paths.addAll(rule.paths());
+        }
+        var path = contentExclusionPath();
+        Files.createDirectories(path.getParent());
+        Files.write(path, paths);
+        contentExclusionsDownloaded = true;
+    }
+
+    /**
+     * Downloads the content exclusions from the GitHub Copilot API and writes them to the global ignores file.
+     * If download is unsuccessful, simply return false.
+     *
+     * @return Whether the content exclusions have been downloaded.
+     */
+    public boolean downloadContentExclusions() {
+        var githubToken = loadGitHubOAuthToken();
+        try {
+            downloadContentExclusions(githubToken);
+            return true;
+        } catch (IOException e) {
+            return false;
+        }
+    }
+
+    /**
+     * Ensures that the content exclusions have been downloaded.
+     *
+     * @return Whether the content exclusions have been downloaded.
+     */
+    private boolean ensureContentExclusionsDownloaded(String githubToken) {
+        if (contentExclusionsDownloaded) return true;
+        try {
+            downloadContentExclusions(githubToken);
+            return true;
+        } catch (IOException e) {
+            ApplicationManager.getApplication().invokeLater(() -> AppMapNotifications.showCopilotExclusionDownloadFailed(e));
+            return false;
+        }
+    }
+
+    /**
+     * @return The path to the global ignores file.
+     */
+    private static Path contentExclusionPath() {
+        return Path.of(SystemProperties.getUserHome(), ".appmap", "navie", "global-ignore");
+    }
+
     /**
      * Creates a new chat session with the GitHub Copilot API.
      * If the OAuth token or the Copilot token could not be retrieved, this method will return {@code null}.
      */
     public @Nullable CopilotChatSession createChatSession() {
         var githubToken = loadGitHubOAuthToken();
-        if (githubToken == null) {
+
+        if (githubToken == null || !ensureContentExclusionsDownloaded(githubToken)) {
```

Comment:
This code is called from several places.
It's also invoked by `appland.copilotChat.ChooseCopilotModelAction`. I'm not sure if we really should make loading&choosing the model fail if the exclusions fail to launch.

Reply:
This isn't ideal, but I think it's acceptable. Most of the time it will work; and if it doesn't, completion is not supposed to work anyway. @kgilpin WDYT?